### PR TITLE
fix: [CI-5023]: Updated artifactory plugin and PLUGIN_FLAT env for artifactory

### PR DESCRIPTION
**Pull request by raghavharness**
## Describe your changes
Updated artifactory plugin and PLUGIN_FLAT env for artifactory
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35569)

### CODE REVIEW

**Comment by raghavharness** [comment]
trigger messagecheck

File: 320-ci-execution/src/main/java/io/harness/ci/buildstate/PluginSettingUtils.java [hunk 1]
```diff
@@ -522,6 +523,7 @@ private static Map<String, String> getUploadToArtifactoryStepInfoEnvVariables(
         resolveStringParameter("sourcePath", "ArtifactoryUpload", identifier, stepInfo.getSourcePath(), true));
     setMandatoryEnvironmentVariable(map, PLUGIN_TARGET,
         resolveStringParameter("target", "ArtifactoryUpload", identifier, stepInfo.getTarget(), true));
+    setMandatoryEnvironmentVariable(map, PLUGIN_FLAT, "true");
```
Comment (shubham149) [inline 320-ci-execution/src/main/java/io/harness/ci/buildstate/PluginSettingUtils.java:526]: This needs to be false

Comment (raghavharness) [reply 320-ci-execution/src/main/java/io/harness/ci/buildstate/PluginSettingUtils.java:526]: @shubham149 This should be true. Checked in Jfrog doc, they have made this false by default. Tested this.
https://www.jfrog.com/confluence/display/CLI/JFrog+CLI

Comment (shubham149) [reply 320-ci-execution/src/main/java/io/harness/ci/buildstate/PluginSettingUtils.java:526]: Can you test once with 1.x version and this version whether upload path and pushed files are same or not in artifactory?

Comment (raghavharness) [reply 320-ci-execution/src/main/java/io/harness/ci/buildstate/PluginSettingUtils.java:526]: yes checked with `plugins/artifactory:1.0.6` image which uses v1 and the path are same.

**Review by shubham149 — COMMENTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35569)* status: 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by shubham149 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35569)* status: :shipit: complete! ✓

**Pull request merged**